ush/python/pygfs/task/jedi.py

```diff
@@ -0,0 +1,276 @@
+        os.symlink(task_config.JEDIEXE, exe_dest)
+
+    @logit(logger)
+    def get_obs_dict(self, task_config: AttrDict[str, Any]) -> Dict[str, Any]:
+        """Compile a dictionary of observation files to copy
+
+        This method extracts 'observers' from the JEDI yaml and from that list, extracts a list of
+        observation files that are to be copied to the run directory
+        from the observation input directory
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        ----------
+        obs_dict: Dict
+            a dictionary containing the list of observation files to copy for FileHandler
+        """
+
+        logger.info(f"Extracting a list of observation files from Jedi config file")
+        observations = find_value_in_nested_dict(self.config, 'observations')
+        logger.debug(f"observations:\n{pformat(observations)}")
+
+        copylist = []
+        for ob in observations['observers']:
+            obfile = ob['obs space']['obsdatain']['engine']['obsfile']
+            basename = os.path.basename(obfile)
+            copylist.append([os.path.join(task_config.COM_OBS, basename), obfile])
+        obs_dict = {
+            'mkdir': [os.path.join(task_config.DATA, 'obs')],
+            'copy': copylist
+        }
+        return obs_dict
+
+    @logit(logger)
+    def get_bias_dict(self, task_config: Dict[str, Any]) -> Dict[str, Any]:
+        """Compile a dictionary of observation files to copy
+
+        This method extracts 'observers' from the JEDI yaml and from that list, extracts a list of
+        observation bias correction files that are to be copied to the run directory
+        from the component directory.
+        TODO: COM_ATMOS_ANALYSIS_PREV is hardwired here and this method is not appropriate in
+        `analysis.py` and should be implemented in the component where this is applicable.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        ----------
+        bias_dict: Dict
+            a dictionary containing the list of observation bias files to copy for FileHandler
+        """
+
+        logger.info(f"Extracting a list of bias correction files from Jedi config file")
+        observations = find_value_in_nested_dict(self.config, 'observations')
+        logger.debug(f"observations:\n{pformat(observations)}")
+
+        copylist = []
+        for ob in observations['observers']:
+            if 'obs bias' in ob.keys():
+                obfile = ob['obs bias']['input file']
+                obdir = os.path.dirname(obfile)
+                basename = os.path.basename(obfile)
+                prefix = '.'.join(basename.split('.')[:-2])
+                for file in ['satbias.nc', 'satbias_cov.nc', 'tlapse.txt']:
+                    bfile = f"{prefix}.{file}"
+                    copylist.append([os.path.join(task_config.COM_ATMOS_ANALYSIS_PREV, bfile), os.path.join(obdir, bfile)])
+                    # TODO: Why is this specific to ATMOS?
```

Comment:
I have an idea to make this agnostic to component in the case bias correction is applied to other components